src/System.Management.Automation/engine/Utils.cs

```diff
@@ -482,9 +482,19 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
 
         internal static string GetApplicationBase(string shellId)
         {
-            // Use the location of SMA.dll as the application base.
-            Assembly assembly = typeof(PSObject).Assembly;
-            return Path.GetDirectoryName(assembly.Location);
+            var baseDirectory = AppContext.BaseDirectory;
+            if (!string.IsNullOrEmpty(baseDirectory))
+            {
+                // need to remove any trailing directory separator characters
+                baseDirectory = baseDirectory.TrimEnd(Path.DirectorySeparatorChar);
+            }
+            else
+            {
+                // host path can be empty if hosted in WinRM, so fallback to using SMA.dll location
+                baseDirectory = Path.GetDirectoryName(typeof(PSObject).Assembly.Location);
+            }
+
+            return baseDirectory;
```

Comment:
[Current AppContext.BaseDirectory implementation](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs,24) return `Assembly.GetEntryAssembly()?.Location` for not AOT (for pwsh.exe it is `pwsh.dll` location). It can be not that we want in some scenarios. Also do we really need `APP_CONTEXT_BASE_DIRECTORY`?
I think logic could be:
```c#
var baseDirectory = Path.GetDirectoryName(typeof(PSObject).Assembly.Location);
if (string.IsNullOrEmpty(baseDirectory))
{
    baseDirectory = Environment.ProcessPath;
}
```